crml/nft/src/lib.rs

```diff
@@ -647,9 +648,15 @@ decl_module! {
 					for (who, entitlement) in listing.royalties_schedule.entitlements.into_iter() {
 						let royalty = entitlement * listing.fixed_price;
 						for_seller -= royalty;
-						imbalance = imbalance.offset(T::MultiCurrency::deposit_into_existing(&who, listing.payment_asset, royalty)?).map_err(|_| Error::<T>::InternalPayment)?;
+						imbalance = match imbalance.offset(T::MultiCurrency::deposit_into_existing(&who, listing.payment_asset, royalty)?) {
+							SameOrOther::Same(value) => value,
+							_ => return Err(Error::<T>::InternalPayment.into()),
```

Comment:
I don't think. an error makes sense anymore, if it's negative or positive it should do the offset all the same.

withdraw creates a negative imbalance
deposit creates a positive imbalance
in the end it imbalance should be `0`